### PR TITLE
Add suggestions on how to use TBB allocator and HugePage support to docs

### LINKED ISSUE #3218: Consider linking `tbbmalloc_proxy`
See https://www.intel.com/content/www/us/en/developer/articles/technical/how-to-use-onetbb-for-memory-allocation-cpp.html and https://oneapi-src.github.io/oneTBB/main/tbb_userguide/automatically-replacing-malloc.html for context.

I tried this very briefly. For the following workload (branch from #3215):

```python
import scipp as sc
table = sc.data.table_xyz(100_000_000)

nbin = 401
x = sc.linspace('x', 0.0, 1.0, num=nbin, unit='m')
y = sc.linspace('y', 0.0, 1.0, num=nbin, unit='m')

for i in range(16):
    table.bin(x=x, y=y).bin(z=100)
```
On my workstation I observe a **10%** speedup... for free (39 seconds instead of 43 seconds).

- Come up with a better set of benchmarks. Is this always better? Where do we get speedups?
- Enable if OS supports it (I think docs say that MacOS does not).

Diff:

```diff
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index d66bcb348..41fe2542a 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -61,7 +61,7 @@ include(GenerateExportHeader)
 
 option(THREADING "Build with multi-threading" ON)
 if(THREADING)
-  find_package(TBB CONFIG REQUIRED)
+  find_package(TBB CONFIG REQUIRED COMPONENTS tbb tbbmalloc_proxy)
   configure_file(
     core/include/scipp/core/parallel-tbb.h core/include/scipp/core/parallel.h
     COPYONLY
diff --git a/lib/core/CMakeLists.txt b/lib/core/CMakeLists.txt
index 47fc4865f..8d95e60d9 100644
--- a/lib/core/CMakeLists.txt
+++ b/lib/core/CMakeLists.txt
@@ -64,7 +64,7 @@ target_link_libraries(
   ${TARGET_NAME} PUBLIC scipp-common scipp-units Boost::Boost Eigen3::Eigen
 )
 if(TBB_FOUND)
-  target_link_libraries(${TARGET_NAME} PUBLIC TBB::tbb)
+  target_link_libraries(${TARGET_NAME} PUBLIC TBB::tbb TBB::tbbmalloc_proxy)
 endif()
 
 # Include tcb/span as system header to avoid compiler warnings.
```

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Closes #3218.

### CODE REVIEW

File: docs/user-guide/tips-tricks-and-anti-patterns.ipynb [hunk 1]
```diff
@@ -413,6 +420,224 @@
...
+    "\n",
+    "Environment variables can be configured for a specific Jupyter kernel.\n",
+    "First, obtain a list of kernels:\n",
+    "\n",
+    "```console\n",
+    "$ jupyter kernelspec list\n",
+    "```\n",
+    "\n",
+    "The list may include, e.g., `/home/<username>/mambaforge/envs/myenv/share/jupyter/kernels/python3`.\n",
+    "Consider making a copy of this folder, e.g., to `/home/<username>/mambaforge/envs/myenv/share/jupyter/kernels/python3-tbbmalloc_proxy`.\n",
```
Comment (jl-wynen) [inline docs/user-guide/tips-tricks-and-anti-patterns.ipynb:493]: This is Linux-specific. 
That being said, does the preload mechanism work on MacOS and Windows?

File: docs/user-guide/tips-tricks-and-anti-patterns.ipynb [hunk 2]
```diff
@@ -413,6 +420,224 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Using explicit HugePages\n",
+    "\n",
+    "HugePages can be reserved explicitly, in contrast to the transparent HugePages described below.\n",
+    "First, check the number of configured HugePages:\n",
+    "\n",
+    "```console\n",
+    "$ cat cat /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages\n",
```
Comment (jl-wynen) [inline docs/user-guide/tips-tricks-and-anti-patterns.ipynb:550]:
```suggestion
    "$ cat /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages\n",
```

**Review by jl-wynen — COMMENTED** [review]
Did you consider using an explicit allocator in C++ instead of setting one globally?

**Comment by SimonHeybrock** [comment]
> Did you consider using an explicit allocator in C++ instead of setting one globally?

I did try and I failed. I ended up with some segfaults that I could not track track (before going with the much simpler solution).

File: docs/user-guide/tips-tricks-and-anti-patterns.ipynb (hunk 1 above)
Comment (SimonHeybrock) [reply docs/user-guide/tips-tricks-and-anti-patterns.ipynb:493]: According to the docs it is possible on Windows, but I do not know how to preload. On MacOS the lib is available, but the docs to not mention support... maybe it is implied in "Linux"?

Comment (SimonHeybrock) [reply docs/user-guide/tips-tricks-and-anti-patterns.ipynb:502]: I added a visible note at the top of the section, marking it is Linux-only.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**